### PR TITLE
Introduce Pipenv for development environment management

### PR DESCRIPTION
**Pull request by ninjinkun**
To improve the CLI tests, I need more pip development module dependencies to this repo (For example, I want to add [responses](https://github.com/getsentry/responses)). However, we don't have any way to force install new dependency to developers. I think it's time to introduce a development environment management.

[Pipenv](https://pipenv.pypa.io/en/latest/) is a popular module that provides development environment management and dependency management. It can create a virtual environment on your local and install dependencies there. If you want to know how to use it, please see new lines on `README.md`.

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+[packages]
+launchable = {editable = true,path = "."}
+
+[scripts]
+build = "python setup.py sdist bdist_wheel"
+install = "pip install -U ."
+test = "python -m nose tests"
+lint = "flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics"
+lint-warn = "flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics"
```
Comment (Konboi) [inline Pipfile:23]: How did you decide these values?

File: Pipfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+flake8 = "*"
+setuptools = ">=30.3.0"
+setuptools-scm = "*"
+wheel = "*"
+autopep8 = "*"
+nose = "*"
+importlib-metadata = "*"
+
+[packages]
+launchable = {editable = true,path = "."}
```
[the remark this answers is not in the document]
Comment (ninjinkun) [reply Pipfile:16]: We have already defined non-dev dependencies at `setup.cfg`. IIUC, they are used to PyPI installation and `pipenv install`.

File: Pipfile (hunk 1 above)
Comment (ninjinkun) [reply Pipfile:23]: I copied these rules from a [GithubActions template](https://github.com/actions/starter-workflows/blob/main/ci/python-package.yml). We already have been using it on our Github Actions. I think we should consider what rule we should use later.

**Review by kohsuke — APPROVED** [review]
I believe we use `pipenv` in a part of mothership.

**Comment by kohsuke** [comment]
Nice to see the development process & environment getting increasingly standardized.

**Review by Konboi — APPROVED** [review]
LGTM

**Pull request merged**